CHANGELOG.md

```diff
@@ -2,160 +2,178 @@
+  made to imitate native mode default behavior by using the `--no-compat` flag.
+
+OCI-mode changes from 3.11 to 4.0 include:
+
+- `run / shell / exec` in OCI-mode now includes support for the following
+  existing CLI flags:
+  - `--add-caps`
+  - `--drop-caps`
+  - `--keep-privs`
+  - `--no-privs`
+  - `--overlay` from directories, bare squashfs and extfs images.
+  - `--workdir`
+  - `--scratch`
+  - `--no-home`
+  - `--no-mount` (dev cannot be disabled in OCI mode
+  - `--no-umask` (with `--no-compat`)
+  - `--writable-tmpfs` (with `--no-compat`)
+- Added `--device` flag to "action" commands (`run`/`exec`/`shell`) when run in
+  OCI mode (`--oci`). Currently supports passing one or more (comma-separated)
+  fully-qualified CDI device names, and those devices will then be made
+  available inside the container.
+- Added `--cdi-dirs` flag to override the default search locations for CDI
+  json files, allowing, for example, users who don't have root access on their
+  host machine to nevertheless create CDI mappings (into containers run with
+  `--fakeroot`, for example).
+- A container run as root, or with `--fakeroot`, has OCI default
+  effective/permitted capabilities.
+- An `--env-file` is evaluated with respect to the host environment, to match
+  native mode behaviour.
+- If the kernel does not support unprivileged overlays, OCI-mode will attempt to
+  use `fuse-overlayfs` and `fusermount` for overlay mounting and unmounting.
+- Support for thee `SINGULARITY_CONTAINLIBS` env var, to specify libraries to
+  bind into `/.singularity.d/libs/` in the container.
+- Support for running OCI-SIF images directly from `docker://`, `http://`,
+  `https://` and `oras://`. URIs.
+- A new `--no-compat` flag can be used with OCI-mode to reduce OCI runtime
+  compatibility and emulate singularity's historic native mode behaviour:
+  - `$HOME`, `/tmp`, `/var/tmp` are bind mounted from the host.
+  - The full `/dev` is bind mounted from the host, unless `mount dev = minimal`
+    in `singularity.conf`.
+  - `bind path` entries in `singularity.conf` are mounted into the container.
+  - The current working directory is mounted into the container, and is the
+    entry point into the container.
+  - The container is read-only unless `--writable-tmpfs` is also used.
+  - The host umask is propagated into the container, unless `--no-umask` is also
+    used.
+  - When a native (non-OCI-SIF) image is run in OCI-mode, environment variables
+    will be shell evaluated on container startup.
+- The `pull` command now accepts a new flag `--oci` for OCI image sources. This
+  will create an OCI-SIF image rather than convert to Singularity's native
+  container format.
+- OCI-SIF images can be pushed/pulled to/from OCI registries as single file
+  artifacts using `oras://` URIs.
+- OCI-SIF images can be pushed/pulled to/from OCI registries as single layer,
+  squashfs layer format OCI images using `docker://` URIs.
```

Comment:
"single layer, squashfs layer format OCI images" - phrasing can maybe be smoothed out?